src/blindpsbt.cpp

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) 2017-2019 The Elements Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <blindpsbt.h>
+
+#include <hash.h>
+#include <primitives/transaction.h>
+#include <primitives/confidential.h>
+#include <psbt.h>
+#include <issuance.h>
+#include <random.h>
+#include <util/system.h>
+
+static secp256k1_context* secp256k1_blind_context = NULL;
+
+class Blind_PSBT_ECC_Init {
+public:
+    Blind_PSBT_ECC_Init() {
+        assert(secp256k1_blind_context == NULL);
+
+        secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
+        assert(ctx != NULL);
+
+        secp256k1_blind_context = ctx;
+    }
+
+    ~Blind_PSBT_ECC_Init() {
+        secp256k1_context *ctx = secp256k1_blind_context;
+        secp256k1_blind_context = NULL;
+
+        if (ctx) {
+            secp256k1_context_destroy(ctx);
+        }
+    }
+};
+
+static Blind_PSBT_ECC_Init ecc_init_on_load;
+
+// Create surjection proof
+bool CreateAssetSurjectionProof(std::vector<unsigned char>& output_proof, const std::vector<secp256k1_fixed_asset_tag>& fixed_input_tags, const std::vector<secp256k1_generator>& ephemeral_input_tags, const std::vector<uint256>& input_asset_blinders, const uint256& output_asset_blinder, const secp256k1_generator& output_asset_tag, const CAsset& asset)
+{
+    int ret;
+    // 1 to 3 targets
+    size_t inputs_to_select = std::min((size_t)3, fixed_input_tags.size());
```

Comment:
Can you replace this with a new constant in blind.h (and find the other place in the wallet that we hardcode 3). We'd like to increase this number at some point, it was chosen to copy Monero in 2015, and they have since increased their number to 11..

Reply:
Done in a new commit.